### PR TITLE
Added script to get the kubeconfig file

### PR DESCRIPTION
**Pull request by e-minguez**

### CODE REVIEW

File: slemicro/README.md [hunk 1]
```diff
@@ -203,3 +207,51 @@ For multiple VMs in parallel:
...
+
+* For Rancher based installations:
+
+```bash
+$ RANCHER=$(./create_vm.sh -f foobar | awk "/After Rancher/ { print $12 }") 
+$ while ! curl -sk ${RANCHER}; do sleep 10; done
+$ KUBECONFIG=$(./get_kubeconfig.sh -f vm-foobar)
+```
+
+* For no Rancher:
```
Comment (atanasdinov) [inline slemicro/README.md:251]:
```suggestion
* For non-Rancher based installations:
```

File: slemicro/get_kubeconfig.sh [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  		KUBECONFIG=/etc/rancher/rke2/rke2.yaml
+  		;;
+		false)
+			die "CLUSTER cannot be false" 2
+			;;
+		*)
+			die "CLUSTER variable not supported" 2
+			;;
+	esac
+	# TODO: remove the hardcoded user
```
Comment (atanasdinov) [inline slemicro/get_kubeconfig.sh:93]: Is this to be addressed later?

File: slemicro/get_kubeconfig.sh [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+	# Check if the commands required exist
+	command -v jq > /dev/null 2>&1 || die "jq not found" 2
+
+	# Login
+	TOKEN=$(curl -sk -X POST https://${VMIP}.sslip.io/v3-public/localProviders/local?action=login -H 'content-type: application/json' -d "{\"username\":\"admin\",\"responseType\": \"token\", \"password\": \"${RANCHERFINALPASSWORD}\"}" | jq -r .token)
+
+	# Get the kubeconfig
+	# TODO: The following API call creates a token that doesn't expire https://github.com/rancher/rancher/issues/37705
+	GENERATEDKUBECONFIG=$(curl -sk "https://${VMIP}.sslip.io/v3/clusters/local?action=generateKubeconfig" -X 'POST' -H 'content-type: application/json' -H "Authorization: Bearer ${TOKEN}")
```
Comment (atanasdinov) [inline slemicro/get_kubeconfig.sh:115]: Do we want to abort with an appropriate message if this fails? Not sure what the output below would be if this is the case.

File: slemicro/get_kubeconfig.sh (hunk 2 above)
Comment (e-minguez) [reply slemicro/get_kubeconfig.sh:93]: Yes

File: slemicro/get_kubeconfig.sh (hunk 3 above)
Comment (e-minguez) [reply slemicro/get_kubeconfig.sh:115]: Probably? Any suggestion? :)

Comment (atanasdinov) [reply slemicro/get_kubeconfig.sh:132]: Leaving this as is for the moment.

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**